util/patch/patch_test.go

```diff
@@ -425,6 +425,10 @@ func TestPatchHelper(t *testing.T) {
 					readyBefore := conditions.Get(obj, clusterv1.ReadyCondition)
 					readyAfter := conditions.Get(objAfter, clusterv1.ReadyCondition)
 
+					// Differences in the LastTransitionTime are intentionally ignored as the LastTransitionTime is re-calculated
+					// in conditions.Set during patch.Helper.Patch() and thus not necessarily equal to the one in the original obj.
+					readyBefore.LastTransitionTime = readyAfter.LastTransitionTime
```

Comment:
Great find! Should we have a way to compare conditions in a better way? cc @fabriziopandini

Reply:
@vincepri @sbueringer in https://github.com/kubernetes-sigs/cluster-api/blob/master/util/conditions/matcher.go we have condition matchers that should already take this into account

Reply:
Perfect, thx